### PR TITLE
Pass BETA_RELEASE to release-upload.sh invocation

### PR DESCRIPTION
**Pull request by mokagio**
See discussion in p1734637168903759/1734636063.032259-slack-C029BMLUGRX

Fixes an issue I introduced by carelessness in https://github.com/Automattic/pocket-casts-ios/pull/2461 when splitting the build from upload script for release builds and forgetting to pass along the `$BETA_RELEASE` env var that CI receives to the new upload script.

## To test

This will be real world tested when a new release will run. As such, I opened this against the latest release branch, so we can check it sooner rather than later.

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary. – N.A.
- [x] I have considered adding unit tests for my changes. – N.A.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics. – N.A.See discussion in p1734637168903759/1734636063.032259-slack-C029BMLUGRX

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/67">7.79 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: .buildkite/commands/release-upload.sh [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash -eu
 
+BETA_RELEASE=${1:-true} # use first call param, default to true for safety
+
+echo "Running $0 with BETA_RELEASE = $BETA_RELEASE..."
```
Comment (mokagio) [inline .buildkite/commands/release-upload.sh:5]: I thought moving the code that reads the call parameters to the start of the script would make the existence and requirement of those parameters clearer.

An alternative would be to remove the fallback value and fail if `$1` is missing. I thought of that option only now, while writing this comment, but I like it more. Better to fail early, IMHO.

I'll leave the PR as is for the moment. If the fail early idea is well received, I'll create a task to roll it out consistently across our repos.

Comment (iangmaia) [reply .buildkite/commands/release-upload.sh:5]: tbh I don't mind either way. Though we've been using in other repos the validation as in `MY_VAR="${1?Error message.}"`, which can also be a convention to follow here.

**Review by iangmaia — APPROVED** [review]
Approving to unblock as the primary issue has been fixed.
Up to you if we should keep the default value or to follow the validation we have in a couple of other repos.

**Comment by AliSoftware** [comment]
Thanks @mokagio for following-up and taking care of this!

**Pull request merged**